### PR TITLE
RHOAI: Test Python client of Model Registry component

### PR DESCRIPTION
**Pull request by rkubis**

### CODE REVIEW

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client pull-ci-opendatahub-io-model-registry-main-e2e-odh-cfo

**Review by isinyaaa — COMMENTED** [review]
Hey Radim, do you have a sample run to share? I'd say this LGTM but it's better to be safe, right :)

File: ci-operator/config/opendatahub-io/model-registry/opendatahub-io-model-registry-main.yaml [hunk 1]
```diff
@@ -38,6 +35,39 @@ tests:
...
+    env:
+      BASE_DOMAIN: openshift-ci-aws.rhaiseng.com
+      HYPERSHIFT_AWS_REGION: us-west-2
+      HYPERSHIFT_NODE_COUNT: "3"
+    test:
+      - as: main
+        cli: latest
+        commands: |
+          # Print last commit of current code
+          git log --name-status HEAD^..HEAD
```
[the remark this answers is not in the document]
Comment (rkubis) [reply ci-operator/config/opendatahub-io/model-registry/opendatahub-io-model-registry-main.yaml]: Oh, thanks! Good tip :) I am going to change it!

**Comment by dhirajsb** [comment]
@rkubis can you not run the Python client test from a pod? The Python APIs are meant to used primarily in Notebooks, which run in pods in RHOAI. 
When the test environment is changed to use Istio Service Mesh, we can use an Authorization token and TLS to connect to the gRPC service either within the service mesh, or through the gateway from outside the service mesh. 
Even when the Python API is changed to use REST, it will need to use the same Authorization token and TLS to connect to the REST service in an Istio configuration.

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client pull-ci-opendatahub-io-model-registry-main-e2e-odh-cfo

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client pull-ci-opendatahub-io-model-registry-main-e2e-odh-cfo

**Comment by rkubis** [comment]
> @rkubis can you not run the Python client test from a pod? The Python APIs are meant to used primarily in Notebooks, which run in pods in RHOAI. When the test environment is changed to use Istio Service Mesh, we can use an Authorization token and TLS to connect to the gRPC service either within the service mesh, or through the gateway from outside the service mesh. Even when the Python API is changed to use REST, it will need to use the same Authorization token and TLS to connect to the REST service in an Istio configuration.

@dhirajsb Thanks for explanation, I thought that Python client is intended to be used from outside world for interacting with Model Registry. Run Python client test from a pod is surely an option here.

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client pull-ci-opendatahub-io-model-registry-main-e2e-odh-cfo

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client pull-ci-opendatahub-io-model-registry-main-e2e-odh-cfo

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client pull-ci-opendatahub-io-model-registry-main-e2e-odh-cfo

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client pull-ci-opendatahub-io-model-registry-main-e2e-odh-cfo

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by tonyxrmdavidson** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Comment by rkubis** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client pull-ci-opendatahub-io-model-registry-main-e2e-odh-mro-optional pull-ci-opendatahub-io-model-registry-main-images periodic-ci-opendatahub-io-model-registry-main-e2e-odh-mro-cron

**Comment by tonyxrmdavidson** [comment]
/pj-rehearse pull-ci-opendatahub-io-model-registry-main-e2e-odh-model-registry-python-client

**Pull request merged**